### PR TITLE
Feat: Written Account Data Component

### LINKED ISSUE #9: Implement entry point to get members and private function to add members
## Description
Implement a public function in the AccountData component to return the members of the account and a private function to add members. This public function will take no arguments and the private function will take an array of contractaddress as an argument

## Execution Steps

- For the entrypoint, define the interface in the `iaccount_data.cairo` file and implement the interface in the `account_data.cairo` file component. Use `#[embeddable_as(AccountDataImpl)]` to define the impl. 
- In the function, initialize an empty array of contractAddress, loop through the members map based on the members count and append each contract address. Return the array of contractaddress
- For the private function, loop through the members arguments, assert that the address is not a zero address, use the member count to add it to the members map and increment the count on each addition.
- Properly document the functions.
- Write tests for the private and the public function and ensure it passes. (use edge cases if possible)

**Note:** Assertion error messages should be defined in the `error.cairo` file and imported to the account file. Members count start from 1.
Before starting this task, please read the contribution guideline properly and follow it thoroughly.

### PR DESCRIPTION
**Pull request by OWK50GA**
## Description 📝
This Pull Request is for implementing the account data component, currently writing the tests for the components

## Related Issues 🔗
Fixes #9 

## Changes Made 🚀
- [x] ✨ Feature Implementation 
- [ ] 🐛 Bug Fix 
- [ ] 📚 Documentation Update 
- [ ] 🔨 Refactoring 
- [ ] ❓ Others (Specify) 

## Checklist ✅
- [x] 🛠 I have tested these changes. 
- [ ] 📖 I have updated the documentation (if applicable). 
- [x] 🎨 This PR follows the project's coding style. 
- [x] 🧪 I have added necessary tests (if applicable).

### CODE REVIEW

File: Scarb.toml [hunk 1]
```diff
@@ -6,9 +6,9 @@ edition = "2024_07"
 # See more keys and their definitions at https://docs.swmansion.com/scarb/docs/reference/manifest.html
 
 [dependencies]
-starknet = "2.8.5"
+starknet = "2.9.2"
```
Comment (codebestia) [inline Scarb.toml:9]: Please use the provided starknet version. Editing this file should not be part of this task.

File: src/account_data.cairo [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
-    use starknet::{ContractAddress};
+    use starknet::ContractAddress;
     #[storage]
     pub struct Storage {
-        transactions: Map<u256, Transaction>,
+        transactions: Map::<u256, Transaction>,
         tx_count: u256, // the transaction length
         threshold: u64, // the number of members required to approve a transaction for it to be executed
-        members: Map<u64, ContractAddress>, // Map(id, member) the members of the account
+        members: Map::<u64, ContractAddress>, // Map(id, member) the members of the account
```
Comment (codebestia) [inline src/account_data.cairo:15]: Use starknet v2.8.5 Map syntax. That is what we are using for now.

**Review by codebestia — CHANGES_REQUESTED** [review]
Hello @OWK50GA Good job so far.
I am hoping your PR will be ready for merge soon.
I added some reviews on this current draft. Please go through it.
